### ISSUE TITLE
Approval of hours

### ISSUE THREAD

**Issue by sondretj** [superprofit/timet-issues#5]
We need to support approval of hours and keeping tracked of approved history. 

AC:
* No invoicing can happen for a given project without all users having approved their timesheet for given month
* Users must be able to track their history of approved hours
* Users must be able to differ between current drafted hours and what has been approved
* Admins must see an overview of approved hours for given month.
* Admins must be able to override approved hours.
* There should be some functionality to approve or submit hours without filling timesheet (users who only submit a total sum each month for example)

More? 

Edit:
* The architecture should support submitting / approving a set of hours individually. Although unlikely,  billable rate may change in the middle of a month for example. (see approval flow that CFO in Sonat showed)
* A snapshot of customer data and / or project data for the linked projects needs to be created as these data may change over time. For example billable rate may have been updated before invoiced, resulting in an invoice with wrong rate.
* UI: Users should see a table of their projects summarized, but also expand each project to see each entry for given period (which will be a given month). 
* UI: Second approval: Admins and other approvers should se a table of projects, the sum of approved hours vs "remaining" and the billable rate of each project. After hours and rate is approved then that project is invoicable.
* As long as second approval has not gone through ( for example), the entries are editable. If entries are edited then the approval is "revoked" or reopened and needs to be restarted

**Comment by henrik1** [comment]
Yeah, I was thinking that we probably don't need to keep two separate books, one with drafts and one approved. Instead, as long as we store the last changed timestamp on all entries, we should be able to approve periods at any given duration (month, week, etc). That would require a new rule though, that you cannot change entries that are before last approval, only append.
